compiler/riscv/codegen/FPTreeEvaluator.cpp

```diff
@@ -316,101 +316,171 @@ OMR::RV::TreeEvaluator::dnegEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    return doublePrecisionUnaryEvaluator(node, TR::InstOpCode::_fsgnjn_d, cg);
    }
 
-static TR::Register *
-conversionHelper(TR::Node *node, TR::InstOpCode::Mnemonic op, TR::CodeGenerator *cg)
-{
+static TR::Register*
+commonFPtoINTconversionEvaluator(TR::Node *node, TR::InstOpCode::Mnemonic op,
+      TR::CodeGenerator *cg)
+   {
+   TR::Node *firstChild = node->getFirstChild();
+   TR::Register *src1Reg = cg->evaluate(firstChild);
+   TR::Register *zero = cg->machine()->getRealRegister(TR::RealRegister::zero);
+   TR::Register *trgReg = cg->allocateRegister(TR_GPR);
+
+   /*
+    * Converting NaN to integral type should produce 0. Since
+    * RISC-V fcvt* instructions don't do it, we have to test
+    * for NaN before and eventually load 0 into result (target)
+    * register.
+    */
+
+   TR::LabelSymbol *startLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *joinLabel = generateLabelSymbol(cg);
+
+   startLabel->setStartInternalControlFlow();
+   joinLabel->setEndInternalControlFlow();
+
+   TR::RegisterDependencyConditions *deps = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, 1, cg->trMemory());
```

Comment:
You also use `src1Reg` during ICF, so it also needs to be added to these dependencies.

Reply:
Right, fixed in 0a505f8